crates/walrus-service/src/node/blob_event_processor.rs

```diff
@@ -212,13 +271,20 @@ pub struct BlobEventProcessor {
     node: Arc<StorageNodeInner>,
 
     // Background processors that process events in parallel.
-    background_processor_senders: Vec<UnboundedSender<(EventHandle, BlobEvent)>>,
+    background_processor_senders: Vec<UnboundedSender<TrackedEvent>>,
     _background_processors: Vec<Arc<JoinHandle<()>>>,
 
     // When there are no background workers, we use a sequential processor to process events using
     // this processor. This is to keep the same behavior as before BackgroundEventProcessor.
     // INVARIANT: sequential_processor must be Some if background_processor_senders is empty.
     sequential_processor: Option<Arc<BackgroundEventProcessor>>,
+
+    // The number of events that are pending to be processed in each background processor.
+    // Each counter is shared with individual BackgroundEventProcessor.
+    //
+    // We use per background processor count to avoid high contention on the Atomic variable when
+    // tracking the total number of pending events to be processed.
+    background_per_processor_pending_event_count: Vec<Arc<AtomicU64>>,
```

Comment:
nit: idk for sure, but would `AtomicU32` be enough, wondering if it saves any cache bandwidth.

Reply:
oh i also wonder if there will be false sharing because all the atomics are packed tightly back to back and end up in the same cache line. Might also be worth testing if adding padding helps with performance.

Reply:
i wonder if it might be useful to have this as a util, something like `ShardedCounter` with api's like `inc`, `dec` and `total`

Reply:
Great suggestion! let me change this to u32 in this PR, and work on the util in a follow up PR.